### ISSUE TITLE
Broken ref for enum entry (doxygen 1.8.18 -> 1.8.19)

### ISSUE THREAD

**Issue by spdw** [doxygen/doxygen#7977]
The following code
```
/*!
 * @defgroup Test Test
 * @{ @brief my group @}
 */

/*!
 * @brief My enum.
 * @ingroup Test
 */
enum class E
{
    A, ///< a documentation.
    B  ///< b documentation.
};

/*!
 * @brief The first link \ref E::A is broken while the second works \ref E::B.
 * @ingroup Test
 */
class C
{
};
```
works in doxygen 1.8.18 and creates the warning
```
test.h:17: warning: unable to resolve reference to 'E::A' for \ref command
```
in Doxygen 1.8.19 and 1.8.20.

**Comment by albert-github** [comment]
The source code is present, but the used doxygen configuration is missing, so:
- Can you please attach a, small, self contained example (source+configuration file in a tar or zip) that allows us to reproduce the problem? Please don't add external links as they might not be persistent.

**Comment by spdw** [comment]
The default configuration (generated with -g) produces the warning

**Comment by albert-github** [comment]
Looks like the second link (E::B) points to the wrong entity though (html/group___test.html).

**Comment by spdw** [comment]
You are right. For the first link (\ref E::A) the link is missing while the second link (\ref E::B) is to the wrong enum value (links to E::A).

**Comment by albert-github** [comment]
Looks like the problem is introduced with the commit:
```
Commit: c0267cafbd0967944de94fbe8f461641a376a561 [c0267ca]
Date: Wednesday, April 15, 2020 7:50:27 PM
Fix for crash in addEnumValuesToEnum() (better solution)
```
the previous version was OK (3901935).

**Comment by spdw** [comment]
Looks like the delayed addition with `extraMembers` might lead to the issue.

**Comment by albert-github** [comment]
It indeed loos like to have to do something with the `extraMembers`.
I did a quick, unsuccessful, test, when I replaced the line: (and remove other references to `extraMembers`)
```
extraMembers->push_back(std::move(fmd));
```
with just
```
mn->push_back(std::move(fmd));
```
your problem runs, but the problem for which the original fix was created (simplified example: https://github.com/doxygen/doxygen/files/5128765/example.tar.gz original problem from the hhvm package as ran on Fossies.org) crashes again.
So a bit a more sophisticated solution than my solution is needed.

**Comment by doxygen** [comment]
Please verify if this commit solves the problem.

**Comment by albert-github** [comment]
I ran
- the example from this Issue
- small "crash" example from the original problem (and added here in comment: https://github.com/doxygen/doxygen/issues/7977#issuecomment-680761939)
- the original "crash" application (hhvm-HHVM 4.53.0)

and the all looked OK.

A fix (63dc5b9b1b3e8fb875304a954e4df934b249034b)  has been integrated in master on github (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
